infra/modules/azure_function_app/function_app.tf

```diff
@@ -50,8 +50,17 @@ resource "azurerm_linux_function_app" "this" {
     length(local.function_app.name) > 32 && !(contains(keys(var.app_settings), "AzureFunctionsWebHost__hostid")) ? { AzureFunctionsWebHost__hostid = "production" } : {},
     var.app_settings,
     local.application_insights.enable ? {
+      # SDK AI Sampling (for dependencies)
       # https://docs.microsoft.com/en-us/azure/azure-monitor/app/sampling
-      APPINSIGHTS_SAMPLING_PERCENTAGE = var.application_insights_sampling_percentage
+      APPINSIGHTS_SAMPLING_PERCENTAGE = var.application_insights_sampling_percentage,
+
+      # Azure Function Host (runtime) AI Sampling (for requests, traces, metrics, etc..)
+      # https://learn.microsoft.com/en-us/azure/azure-functions/configure-monitoring?tabs=v2#overriding-monitoring-configuration-at-runtime
+      AzureFunctionsJobHost__logging__applicationInsights__samplingSettings__minSamplingPercentage     = var.application_insights_sampling_percentage,
```

Comment:
do we need `isEnabled=true` as well?

Reply:
If we want to let appinsights be enabled by default, sure

Reply:
it's enabled by default so we don't need to